shard_consumer.go

```diff
@@ -194,7 +195,9 @@ mainloop:
 					origErrStr = fmt.Sprintf("(%s) ", awsErr.OrigErr())
 				}
 				k.config.logger.Log("Got error: %s %s %sretry count is %d / %d", awsErr.Code(), awsErr.Message(), origErrStr, retryCount, maxErrorRetries)
-				if retryCount < maxErrorRetries {
```

Comment:
Thought, since we can separate throttles from "codes which are retry-able without any further action." maybe we should throttle and retry count differently for those? For example never stop retrying on throttles maybe? Just a thought, not a blocker.